### PR TITLE
Pack display name validation

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28078

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ChanochShayner** [comment]
Because of big changes in the SDK that caused conflicts, I opened the new PR, sorry and thanks!
Closing this one - https://github.com/demisto/demisto-sdk/pull/1095

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 1]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
                 is_valid = self._are_playbook_entities_versions_valid(playbook_data, file_path)
 
         return is_valid
+
+    def _is_pack_display_name_already_exist(self, pack_metadata_data):
+        """Check if the pack display name already exists
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/id.py:405]: exists where - be explicit

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 2]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
...
+        Args:
+            pack_metadata_data (dict): Dictionary that holds the extracted details from the given metadata file.
+        Returns:
+            bool. Whether the metadata file is valid or not.
+        """
+        new_pack_id = list(pack_metadata_data.keys())[0]
+        new_pack_name = pack_metadata_data[new_pack_id]['name']
+        for pack_id, pack_data in self.packs_set.items():
+            if new_pack_name == pack_data['name'] and new_pack_id != pack_id:
+                return False, Errors.pack_name_already_exists(new_pack_name)
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/id.py:415]: is the other way around (id exists but has a different name) legal? should we check for that?

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 3]
```diff
@@ -409,6 +409,11 @@ def are_valid_files(self) -> str:
         # We only check pack dependencies for -g flag
         if self.validate_dependencies:
             self.validate_pack_dependencies()
+        if id_set_validations:
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:412]: drop a line and add a comment about this section.

File: demisto_sdk/commands/validate/validate_manager.py [hunk 4]
```diff
@@ -692,7 +692,8 @@ def validate_pack_unique_files(self, pack_path: str, pack_error_ignore_list: dic
                                                                skip_id_set_creation=self.skip_id_set_creation,
                                                                prev_ver=self.prev_ver,
                                                                json_file_path=self.json_file_path)
-        pack_errors = pack_unique_files_validator.are_valid_files()
+        pack_errors = pack_unique_files_validator.are_valid_files(self.id_set_validations,
+                                                                  pack_error_ignore_list)
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:696]: no need to pass the pack ignore list - it is passed in the `PackUniqueFilesValidator` setup.
internally simply use the `self.ignored_errors`.

**Review by roysagi — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 5]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
...
+
+    def is_unique_file_valid_in_set(self, pack_path, ignored_errors=None):
+        """Check if the unique file is valid in the id_set
+                Args:
+                    pack_path (string): Path to the file.
+                    ignored_errors (list): a list of ignored errors for the specific file
+                Returns:
+                    bool. Whether the file is valid in the id_set or not.
+                    string. Error massage if the file is invalid else None.
+                """
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:426]: done

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 1 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:405]: done

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 2 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:415]: The pack_id is the name of the folder in our repo (there can be no duplicates)
pack_name - The display name of the pack.
The specific pack we do validation is also in id_set and therefore id_set already has its name, so we make sure that there is no other pack with the same name.

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py (hunk 3 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:412]: done

File: demisto_sdk/commands/validate/validate_manager.py (hunk 4 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/validate/validate_manager.py:696]: done

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 2 above)
Comment (roysagi) [reply demisto_sdk/commands/common/hook_validations/id.py:415]: then maybe we should change `pack_id` to `pack_folder_name` to clarify this.

Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py]: done.

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 6]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
...
+            bool. Whether the metadata file is valid or not.
+        """
+        new_pack_folder_name = list(pack_metadata_data.keys())[0]
+        new_pack_name = pack_metadata_data[new_pack_folder_name]['name']
+        for pack_folder_name, pack_data in self.packs_set.items():
+            if new_pack_name == pack_data['name'] and new_pack_folder_name != pack_folder_name:
+                return False, Errors.pack_name_already_exists(new_pack_name)
+        return True, None
+
+    def is_unique_file_valid_in_set(self, pack_path, ignored_errors=None):
```
Comment (guykeller) [inline demisto_sdk/commands/common/hook_validations/id.py:418]: Unique how? Valid in what way? Is it a file or necessarily a pack?
This name is a bit confusing to me, I suggest reconsidering it

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 7]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
...
+        """Check if the pack display name already exists in our repo
+        Args:
+            pack_metadata_data (dict): Dictionary that holds the extracted details from the given metadata file.
+        Returns:
+            bool. Whether the metadata file is valid or not.
+        """
+        new_pack_folder_name = list(pack_metadata_data.keys())[0]
+        new_pack_name = pack_metadata_data[new_pack_folder_name]['name']
+        for pack_folder_name, pack_data in self.packs_set.items():
+            if new_pack_name == pack_data['name'] and new_pack_folder_name != pack_folder_name:
```
Comment (guykeller) [inline demisto_sdk/commands/common/hook_validations/id.py:414]: Can you explain the second condition here in details please?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 8]
```diff
@@ -409,6 +409,13 @@ def are_valid_files(self) -> str:
         # We only check pack dependencies for -g flag
         if self.validate_dependencies:
             self.validate_pack_dependencies()
+
+        # Check if unique files valid in ID set
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:413]: done

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 9]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
...
+                return False, Errors.pack_name_already_exists(new_pack_name)
+        return True, None
+
+    def is_unique_file_valid_in_set(self, pack_path, ignored_errors=None):
+        """Check if the unique file is valid in the id_set
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:419]: done

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 10]
```diff
@@ -397,3 +400,37 @@ def is_file_valid_in_set(self, file_path, file_type, ignored_errors=None):
...
         return is_valid
+
+    def _is_pack_display_name_already_exist(self, pack_metadata_data):
+        """Check if the pack display name already exists in our repo
+        Args:
+            pack_metadata_data (dict): Dictionary that holds the extracted details from the given metadata file.
+        Returns:
+            bool. Whether the metadata file is valid or not.
+        """
+        new_pack_folder_name = list(pack_metadata_data.keys())[0]
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:411]: https://github.com/demisto/demisto-sdk/pull/1122#discussion_r579768185

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 7 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:414]: https://github.com/demisto/demisto-sdk/pull/1122#discussion_r577585476

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 6 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/id.py:418]: Regular files - 
image
Unique files - 
image
We have `is_file_valid_in_set` function for the regular files
and I added `is_unique_file_valid_in_set ` function

**Review by guykeller — APPROVED** [review]

**Comment by guykeller** [comment]
@ChanochShayner Please fix coverage and we're good to go

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 * Added the **doc-review** command to check spelling in .md and .yml files as well as a basic release notes review.
+* Added a validation that pack display name did not exist in our repo.
```
Comment (roysagi) [inline CHANGELOG.md:3]:
```suggestion
* Added a validation that a pack's display name does not already exist in content repository.
```

File: demisto_sdk/commands/common/errors.py [hunk 12]
```diff
@@ -1182,6 +1183,12 @@ def pack_metadata_non_approved_tags(non_approved_tags: set) -> str:
     def pack_metadata_price_change(old_price, new_price) -> str:
         return f"The pack price was changed from {old_price} to {new_price} - revert the change"
 
+    @staticmethod
+    @error_code_decorator
+    def pack_name_already_exists(new_pack_name) -> str:
+        return f"The name of your pack is: {new_pack_name} but there is an existing pack with that name, " \
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1189]:
```suggestion
        return f"A pack named: {new_pack_name} already exists in content repository, " \
```

File: demisto_sdk/commands/common/errors.py [hunk 13]
```diff
@@ -1182,6 +1183,12 @@ def pack_metadata_non_approved_tags(non_approved_tags: set) -> str:
...
+               f"please change the pack's name in the metadata file."
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1190]:
```suggestion
               f" change the pack's name in the metadata file."
```

**Review by roysagi — APPROVED** [review]
@ChanochShayner - Please see my notes before merging.

File: CHANGELOG.md (hunk 11 above)
Comment (ChanochShayner) [reply CHANGELOG.md:3]: done

File: demisto_sdk/commands/common/errors.py (hunk 12 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py:1189]: done

File: demisto_sdk/commands/common/errors.py (hunk 13 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py:1190]: done

**Pull request merged**